### DISCUSSION TITLE
Pytorch model does not seems to return the correct result

### DISCUSSION THREAD

**Discussion by carlosuc3m** [bytedeco/javacpp-presets#1323] [Q&A] [answered]
Hello everyone,
I am working with the Pytorch presets trying to load and execute a model saved in torchscript jit format.
However the resulting tensor is not what it should be expected.

This is the code that I am using: 

```

public static void main(String[] args) {
    	// Load a Torchscript model
    	String modelSource = "weights-torchscript.pt";
    	JitModule model = org.bytedeco.pytorch.global.torch.load(modelSource);
    	
    	// Create an input tensor
    	org.bytedeco.pytorch.Tensor inputTensor = 
    			org.bytedeco.pytorch.Tensor.create(new float[512* 512], new long[] {1, 1, 512, 512});
    	
    	// Run the model
		model.eval();
    	IValue outputTensor = model.forward(new IValueVector(new IValue(inputTensor)));
    	
    	// Print shape of the output tensor
    	System.out.println(Arrays.toString(outputTensor.toTensor().shape()));
    	
    	// Retrieve result
    	float[] array = new float[512 * 512 * 2];
    	outputTensor.toTensor().data_ptr_float().get(array);
    	
    	System.out.println("Done");
    }

```

The produced `outputTensor` only has to unique values repeated , and it shouldn't be the case.

Is that the correct way to load and execute a model? What am I missing?
Regards,
Carlos

**Comment by saudet** [comment]
Can you try to call createIndexer() instead of data_ptr_float()?

**Reply by carlosuc3m** [reply]
Same result. That model process a single channel input into a 2-channel output. All the pixels in each of teh channels are the same, and it should not be like that

**Reply by saudet** [reply]
What kind of result are you expecting when all the elements of the input are zeros?

**Reply by carlosuc3m** [reply]
more or less random noise.
If you run the equivalent code in python the result is different

```
import torch
model = torch.jit.load('/content/drive/MyDrive/model/weights-torchscript.pt')
model.eval()
input = torch.zeros(1,1, 512, 512)
out = model(input)
print(out.unique().shape)
```

the output is:
```
torch.Size([95491])
```
95491 different values with respect to the only 2 obtained when running it in Java.

For reference teh model used is:
https://zenodo.org/record/6811922/files/weights-torchscript.pt?download=1

**Reply by saudet** [reply]
Please try again with the snapshots: http://bytedeco.org/builds/

**Reply by carlosuc3m** [reply]
It still does not work.
Here are the dependencies used:

```
    <dependencies>
            <groupId>org.bytedeco</groupId>
            <artifactId>pytorch-platform</artifactId>
            <version>1.13.1-1.5.9-SNAPSHOT</version>
        </dependency>
		<dependency>
            <groupId>org.bytedeco</groupId>
            <artifactId>mkl-platform-redist</artifactId>
            <version>2022.2-1.5.9-SNAPSHOT</version>
        </dependency>
    </dependencies>
```

**Reply by saudet** [reply]
Does it do the same in C++ without JavaCPP? It's really not complicated code, so I'm guessing this is an issue with the C++ API of PyTorch, if anything.

**Reply by carlosuc3m** [reply]
I will check it and get back with the result

**Reply by carlosuc3m** [reply, marked as answer]
I was wrong the whole time the results were perfect. After checking against the python implementation I realised that the results were fine all the time.
